components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAO.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.device.dao;
+
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.device.model.DeviceFlowDO;
+
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * New set of DAO classes  for each purpose  and factory class to get instance of each DAO classes were introduced
+ * during  this step.
+ */
+public interface DeviceFlowDAO {
+
+    /**
+     * This will be used to enter the value to the database tables.
+     *
+     * @param deviceCode  Code that is used to identify the device
+     * @param userCode    Code that is used to correlate user and device
+     * @param consumerKey Consumer key of the client application
+     * @param scope       Requesting scopes
+     * @param expiresIn   Device code valid period
+     * @throws IdentityOAuth2Exception
+     */
+    void insertDeviceFlow(String deviceCode, String userCode, String consumerKey, String scope, Long expiresIn,
+                          int interval) throws
+            IdentityOAuth2Exception;
+
+    /**
+     * Get the client id that has involved with user code.
+     *
+     * @param userCode Code that is used to correlate user and device
```

Comment:
Add a full stop at the end of the comment... check for other places in the PR as well